examples/vision/text_image_denoiser.py

```diff
@@ -0,0 +1,295 @@
+- [Enhancing OCR Accuracy with Super-Resolution](https://cdn.iiit.ac.in/cdn/cvit.iiit.ac.in/images/ConferencePapers/2018/ocr_Ankit_Lat_ICPR_2018.pdf)
+
+- [Improving the Perceptual Quality of Document Images Using Deep Neural Network](http://mile.ee.iisc.ac.in/publications/softCopy/DocumentAnalysis/ISNN_11page_65.pdf)
+"""
+
+"""
+## Additional Set-up
+"""
+
+# !pip install pybind11
+# !pip install fastwer
+# !pip install pytesseract
+# !sudo apt install tesseract-ocr
+
+import fastwer
+import pytesseract
+import numpy as np
+from glob import glob
+import matplotlib.pyplot as plt
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras.applications import resnet_v2
+
+"""
+## Dataset
+We have created a synthetic document dataset that depicts several real world document
+noises; we have used Blurs and Morphological filters to reconstruct real world noises.
+"""
+
+"""shell
+!gdown https://drive.google.com/uc?id=1jQfHJUUQcpktQcjIzitvb4SldKwqcAMr
+!unzip -q data.zip
+"""
+
+"""
+## Dataset Pipeline
+"""
+
+train_source = sorted(glob("data/source/*.jpg"))[:-80]
+train_targets = sorted(glob("data/target/*.jpg"))[:-80]
+
+val_source = sorted(glob("data/source/*.jpg"))[-80:]
+val_targets = sorted(glob("data/target/*.jpg"))[-80:]
+
+HEIGHT, WIDTH = 352, 608
+BATCH_SIZE = 4
+BUFFER_SIZE = int(len(train_source) * 0.2)
+
+
+def preprocess(image):
+    image = tf.io.decode_png(tf.io.read_file(image), channels=3)
+    image = tf.image.resize_with_pad(
+        image=image, target_height=HEIGHT, target_width=WIDTH
+    )
+    image = resnet_v2.preprocess_input(image)
+    return image
+
+
+def data_preprocess(source, target):
+    source_image = preprocess(source)
+    target_image = preprocess(target)
+    return source_image, target_image
+
+
+def denormalize(img_array):
+    img_array += 1
+    img_array *= 127.5
+    return img_array
+
+
+train_set = tf.data.Dataset.from_tensor_slices((train_source, train_targets))
+train_set = train_set.map(data_preprocess, tf.data.AUTOTUNE).shuffle(BUFFER_SIZE)
+train_set = train_set.batch(BATCH_SIZE).repeat()
+
+valid_set = tf.data.Dataset.from_tensor_slices((val_source, val_targets))
+valid_set = valid_set.map(data_preprocess, tf.data.AUTOTUNE).batch(BATCH_SIZE)
+
+"""
+## Dataset Visualiztion
+"""
+
+fig, ax = plt.subplots(1, 2, figsize=(16, 8))
+ax = ax.flatten()
+for i, f in enumerate(valid_set.take(1)):
+    noise, clean = f
+    noise = denormalize(noise[i].numpy())
+    clean = denormalize(clean[i].numpy())
+    ax[0].imshow(noise.astype(np.int))
+    ax[0].set_xlabel("source")
+    ax[1].imshow(clean.astype(np.int))
+    ax[1].set_xlabel("target")
+fig.text(x=0.45, y=0.78, s="Dataset Visualization", fontweight="bold")
+fig.set_tight_layout(None)
+
+"""
+## Model
+
+We have used Pre-trained ```ResNet50V2``` U-Net as Encoder, The Feature Maps are
+extracted at different levels forming Downscaling path, Then ```Sub-Pixel Layers``` on
+the Upscaling path with skip-connections.
+"""
```

Comment:
@anish9 ( cc @fchollet )
I think the model section still needs more description to explain each of its components. 

Apart from that, please make a long sentence into smaller sentences if possible. Avoid commas to merge multiple sentences. Also, use a different paragraph to explain the decoder and encoder for better readability. For example,

> We have used ResNet-50 backbone as the encoder of U-Net architecture. ....
....

> For the decoder, we have used ....
....

Additionally,  if possible use a model diagram to demonstrate visually.

Reply:
@innat, is this fine

We have used Encoder and Decoder Components for constructing our U-Net block. In the Encoder part we extract Feature maps at different scales from the pre-trained ResNet50V2 model forming the Downscaling path in U-Net. In the Decoder part we use Skip-Connection and Sub-Pixel Layer forming the Upscaling Path of U-Net. The Model on its Design aspect is kept light-weight with just 2.7 Million parameters.

Reply:
LGTM. Also, please avoid sudden uppercase, words like - Components, Feature, Model, Million, Desing, etc. There are probably some places where it happens throughout the file.